apps/webapp/app/routes/api.v1.runs.$runId.tasks.ts

```diff
@@ -233,16 +238,21 @@ export class RunTaskService {
         status = "CANCELED";
       } else {
         status =
-          (taskBody.delayUntil && taskBody.delayUntil.getTime() > Date.now()) || taskBody.trigger
+          delayUntilInFuture || callbackEnabled || taskBody.trigger
             ? "WAITING"
             : taskBody.noop
             ? "COMPLETED"
             : "RUNNING";
       }
 
+      const taskId = ulid();
+      const callbackUrl = callbackEnabled
+        ? `${env.APP_ORIGIN}/api/v1/runs/${runId}/tasks/${taskId}/callback/${generateSecret()}`
```

Comment:
Could you update the `generateSecret` util function to receive a param with number of bytes (defaulted to 32) and pass in 12 here? The URLs are a little long using a 32 byte hex secret.